src/main/java/org/apache/cassandra/distributed/api/SimpleQueryResult.java

```diff
@@ -89,6 +99,12 @@ private SimpleQueryResult(String[] names, Object[][] results, Predicate<Row> fil
         return Collections.unmodifiableList(Arrays.asList(names));
     }
 
+    @Override
+    public List<String> warnings()
+    {
+        return Collections.unmodifiableList(Arrays.asList(warnings));
```

Comment:
Nit: Shuold we maybe store it in the list right away? I'm not sure what's the benefit of having an array if expose a list in API anyways. Additional benefit is consistency with a builder!

Reply:
No objections from me.